src/tablenavigation.js

```diff
@@ -0,0 +1,480 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module table/tablenavigation
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import { getSelectedTableCells, getTableCellsContainingSelection } from './utils';
+import { findAncestor } from './commands/utils';
+import TableWalker from './tablewalker';
+import Rect from '@ckeditor/ckeditor5-utils/src/dom/rect';
+import ModelRange from '@ckeditor/ckeditor5-engine/src/model/range';
+import TreeWalker from '@ckeditor/ckeditor5-engine/src/model/treewalker';
+import { keyCodes } from '@ckeditor/ckeditor5-utils/src/keyboard';
+import priorities from '@ckeditor/ckeditor5-utils/src/priorities';
+
+/**
+ * This plugin enables a keyboard navigation for tables.
+ * It is loaded automatically by the {@link module:table/table~Table} plugin.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class TableNavigation extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'TableNavigation';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const view = this.editor.editing.view;
+		const viewDocument = view.document;
+
+		// Handle Tab key navigation.
+		this.editor.keystrokes.set( 'Tab', ( ...args ) => this._handleTabOnSelectedTable( ...args ), { priority: 'low' } );
+		this.editor.keystrokes.set( 'Tab', this._getTabHandler( true ), { priority: 'low' } );
+		this.editor.keystrokes.set( 'Shift+Tab', this._getTabHandler( false ), { priority: 'low' } );
+
+		// Note: This listener has the "high+1" priority because we would like to avoid collisions with other features
+		// (like Widgets), which take over the keydown events with the "high" priority. Table navigation takes precedence
+		// over Widgets in that matter (widget arrow handler stops propagation of event if object element was selected
+		// but getNearestSelectionRange didn't returned any range).
+		this.listenTo( viewDocument, 'keydown', ( ...args ) => this._onKeydown( ...args ), { priority: priorities.get( 'high' ) + 1 } );
+	}
+
+	/**
+	 * Handles {@link module:engine/view/document~Document#event:keydown keydown} events for the <kbd>Tab</kbd> key executed
+	 * when the table widget is selected.
+	 *
+	 * @private
+	 * @param {module:utils/eventinfo~EventInfo} eventInfo
+	 * @param {module:engine/view/observer/domeventdata~DomEventData} domEventData
+	 */
+	_handleTabOnSelectedTable( domEventData, cancel ) {
+		const editor = this.editor;
+		const selection = editor.model.document.selection;
+
+		if ( !selection.isCollapsed && selection.rangeCount === 1 && selection.getFirstRange().isFlat ) {
```

Comment:
I think we can easily drop this `if()` because `Selection#getSelectedElement()` does it internally anyway.

Reply:
Methods `_handleTabOnSelectedTable` and `_getTabHandler` are just moved from `TableEditing` without any changes.